managed/models/agent_model.go

```diff
@@ -340,8 +342,12 @@ func (s *Agent) DSN(service *Service, dsnParams DSNParams, tdp *DelimiterPair) s
 		cfg.Params = make(map[string]string)
 		if s.TLS {
 			// It is mandatory to have "custom" as the first case.
+			// Except case for backward compatibility.
 			// Skip verify for "custom" is handled on pmm-agent side.
 			switch {
+			// Backward compatibility
+			case pmmAgentVersion != nil && s.TLSSkipVerify && pmmAgentVersion.Less(v2_42):
```

Comment:
when we can have pmmAgentVersion nil?
```suggestion
			case s.TLSSkipVerify && (pmmAgentVersion == nil || pmmAgentVersion.Less(v2_42)):
```

Reply:
@BupycHuk In tests https://github.com/percona/pmm/pull/3014/files#diff-e8c710b75bb00d5db0460e38961c6d4bd9c63fa549791eb9d091057a387f9b22R281 since there is new param. Or you want to extend them with specific versions?